static/app/utils/timeSeries/getIntervalForTimeSeriesQuery.tsx

```diff
@@ -0,0 +1,35 @@
+import sortBy from 'lodash/sortBy';
+
+import type {DateTimeObject} from 'sentry/components/charts/utils';
+import {parseFunction} from 'sentry/utils/discover/fields';
+import {intervalToMilliseconds} from 'sentry/utils/duration/intervalToMilliseconds';
+import {getIntervalForMetricFunction} from 'sentry/views/insights/database/utils/getIntervalForMetricFunction';
+
+/**
+ * Given a list of requested Y axes and a date range, figures out the most appropriate interval.
+ *
+ * This is a legacy function that was useful in the days of span metrics, where some metrics were much more expensive to query than others. In the age of EAP, this is not needed anymore. In EAP, granularity is _very cheap_, and we can be more generous with it regardless of the specific metric being queried.
```

Comment:
Hopefully we can make this comment moot soon by increasing granularity everywhere, which would be a huge improvement in experience!